### DISCUSSION TITLE
New Updates to the library

### DISCUSSION THREAD

**Discussion by limistah** [limistah/react-here-map#15] [Announcements]
Hi there,

I plan to add a new update to the library.

This would include:

- Total rewrite of the library (in ts)
- Bug fixes and improvements
- Improved DX

**Comment by udrSilva** [comment]
is there any way to catch the map view change? something like onMapViewChange?

**Reply by limistah** [reply]
What kind of change are you trying to capture?

as an example, when the props to the react element(s) changes?

**Reply by udrSilva** [reply]
I’m trying to load map pins from the backend api so if user zoom out the map i want to get the map boundary coords to send to backend. 

Or if user search for an address by zoom in or move the map, needs to refresh the pins.

**Reply by udrSilva** [reply]
Also noticed engine type and map type props are not working. But not sure my configuration was the issue. If you have any examples please share 🙌🏻

**Reply by limistah** [reply]
Does the events not suit this use case?

https://www.here.com/docs/bundle/maps-api-for-javascript-developer-guide-3.2/page/topics/map-events.html

**Reply by limistah** [reply]
> Also noticed engine type and map type props are not working.

Please can you share a sample code?

**Reply by udrSilva** [reply]
> Does the events not suit this use case?
> 
> https://www.here.com/docs/bundle/maps-api-for-javascript-developer-guide-3.2/page/topics/map-events.html

Yes events can handle that (can we use events with wrapper?). Isn’t it much easier if there was callback function to return lat long and zoom values after map changed.

**Reply by udrSilva** [reply]
> > Also noticed engine type and map type props are not working.
> 
> Please can you share a sample code?

Sure Will do tomorrow it’s 11.40PM for me.

**Reply by limistah** [reply]
https://github.com/limistah/react-here-map/discussions/15#discussioncomment-15222362

Yeah, that can come as an update to the library.
